src/vscode-bicep/src/test/e2e/runTests.ts

```diff
@@ -3,45 +3,62 @@
 import * as path from "path";
 import * as cp from "child_process";
 import * as os from "os";
+import { minVersion } from "semver";
 import {
   runTests,
   downloadAndUnzipVSCode,
   resolveCliPathFromVSCodeExecutablePath,
 } from "vscode-test";
 
+import packageJson from "../../../package.json";
+
 async function go() {
   try {
-    const vscodeExecutablePath = await downloadAndUnzipVSCode("stable");
-    const cliPath = resolveCliPathFromVSCodeExecutablePath(
-      vscodeExecutablePath
-    );
-
-    const isRoot = os.userInfo().username === "root";
-
-    // some of our builds run as root in a container, which requires passing
-    // the user data folder relative path to vs code itself
-    const userDataDir = "./.vscode-test/user-data";
-    const userDataArguments = isRoot ? ["--user-data-dir", userDataDir] : [];
-
-    const extensionInstallArguments = [
-      "--install-extension",
-      "ms-dotnettools.vscode-dotnet-runtime",
-      ...userDataArguments,
-    ];
-
-    // Install .NET Install Tool as a dependency.
-    cp.spawnSync(cliPath, extensionInstallArguments, {
-      encoding: "utf-8",
-      stdio: "inherit",
-    });
-
-    await runTests({
-      vscodeExecutablePath,
-      extensionDevelopmentPath: path.resolve(__dirname, "../../.."),
-      extensionTestsPath: path.resolve(__dirname, "index"),
-      extensionTestsEnv: { NODE_ENV: "test" },
-      launchArgs: ["--enable-proposed-api", ...userDataArguments],
-    });
+    const minSupportedVSCodeSemver = minVersion(packageJson.engines.vscode);
+
```

Comment:
Reading and extracting the version in GH actions and ADO pipeline is not straightforward at all and super hacky, so I gave up that approach and ended up updating this test runner.

Reply:
Sounds good. Personally I like this approach anyway as it's more portable